### PR TITLE
Activity Selection Psuedocode

### PR DESCRIPTION
**Pull request by pjzzz**
Pseudocode for Activity Selection Problem. #13

### CODE REVIEW

File: Pseudocode/Greedy Algorithms/SourceCode.tex [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    \Ensure One Based Indexing for the Array
+    \Statex
+    \Function{Greedy-Activity-Selector}{$Activity, start, finish$}
+        \State Sort Activities into monotonically increasing order by finish time
+        \State $n \gets start.length$
+        \State $Selected \gets \{Activity[1]\}$  
+        \State $j \gets 1$ 
+
+        \For{$ i = 2 : n$}
+            \If{$start[i] >= finish[j]$}
```
Comment (ghost) [inline Pseudocode/Greedy Algorithms/SourceCode.tex]: Use `\geq`

File: Pseudocode/Greedy Algorithms/SourceCode.tex [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  \begin{algorithmic}[1]
+    \Ensure One Based Indexing for the Array
+    \Statex
+    \Function{Greedy-Activity-Selector}{$Activity, start, finish$}
+        \State Sort Activities into monotonically increasing order by finish time
+        \State $n \gets start.length$
+        \State $Selected \gets \{Activity[1]\}$  
+        \State $j \gets 1$ 
+
+        \For{$ i = 2 : n$}
```
Comment (ghost) [inline Pseudocode/Greedy Algorithms/SourceCode.tex]: Use `\textbf{to}` instead of `:`. Ensure proper spacing

**Review by ghost — CHANGES_REQUESTED** [review]

**Comment by ghost** [comment]
I think it's better not to keep the `start` and `finish` time as separate arrays (as it would create an issue while sorting). My suggestion is to pass a single parameter `Activity` and mention in a comment that `Activity[i]` contains the start and finish time of the `i-th` activity.

Then, you can access the `start` time by `Activity[i].start` .

Please make these changes in your code.

File: Pseudocode/Greedy Algorithms/SourceCode.tex [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+            \If{$start[i] >= finish[j]$}
+                \State $Selected \gets Selected  \cup \{Activity[i]\} $
+                \State $j \gets i$
+            \EndIf
+        \EndFor
+
```
Comment (ghost) [inline Pseudocode/Greedy Algorithms/SourceCode.tex]: Include a blank line by `\Statex`

**Review by ghost — CHANGES_REQUESTED** [review]

File: Pseudocode/Greedy Algorithms/SourceCode.tex [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+% Remove the Numbering of the Algorithm
+\usepackage{caption}
+\DeclareCaptionLabelFormat{algnonumber}{Algorithm}
+\captionsetup[algorithm]{labelformat = algnonumber}
+
+\begin{document}
+
+\begin{algorithm}
+
+  \caption{Activity Selection Problem}
```
Comment (ghost) [inline Pseudocode/Greedy Algorithms/SourceCode.tex]: Replace it with `\caption{Maximize the number of activities without overlapping schedules}`

File: Pseudocode/Greedy Algorithms/SourceCode.tex [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+  \begin{algorithmic}[1]
+    \Ensure One Based Indexing for the Array
+    \Statex
+    \Function{Greedy-Activity-Selector}{$Activity, start, finish$}
+        \State Sort Activities into monotonically increasing order by finish time
```
Comment (ghost) [inline Pseudocode/Greedy Algorithms/SourceCode.tex]: I think it should be `non decreasing order`? Correct me if I am wrong

**Review by ghost — CHANGES_REQUESTED** [review]

File: Pseudocode/Greedy Algorithms/SourceCode.tex (hunk 3 above)
Comment (ghost) [reply Pseudocode/Greedy Algorithms/SourceCode.tex]: Include a blank line here

File: Pseudocode/Greedy Algorithms/Activity Selection/SourceCode.tex [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+
+\begin{algorithm}
+
+  \caption{Maximize the number of activities without overlapping schedules}
+  \begin{algorithmic}[1]
+    \Ensure One Based Indexing for the Array
+    \Statex
+    \Function{Greedy-Activity-Selector}{$Activity$}
+        \LeftComment{$Activity[i]$ contains the $start$ and $finish$ time of $i^{th}$ Activity}
+
```
Comment (ghost) [inline Pseudocode/Greedy Algorithms/Activity Selection/SourceCode.tex]: Include a blank line by `\Statex`

File: Pseudocode/Greedy Algorithms/Activity Selection/SourceCode.tex [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+        \State $j \gets 1$ 
+
+        \For{$ i = 2 \textbf{ to } n$}
+            \If{$Activity[i].start \geq Activity[j].finish$}
+                \State $Selected \gets Selected  \cup \{Activity[i]\} $
+                \State $j \gets i$
+                
+            \EndIf
+        \EndFor
+
```
Comment (ghost) [inline Pseudocode/Greedy Algorithms/Activity Selection/SourceCode.tex:45]: Add a blank line by `\Statex`

**Review by ghost — CHANGES_REQUESTED** [review]

**Comment by ghost** [comment]
Everything looks good. Can you just add the 2 blank lines that I've mentioned?

File: Pseudocode/Greedy Algorithms/SourceCode.tex (hunk 5 above)
Comment (pjzzz) [reply Pseudocode/Greedy Algorithms/SourceCode.tex]: According to Cormen and [Wikipedia](https://en.wikipedia.org/wiki/Activity_selection_problem#Explanation) it should be monotonically increasing order.

Comment (ghost) [reply Pseudocode/Greedy Algorithms/SourceCode.tex]: Oh, I guess that's it then. Please edit it to the original form!

Comment (ghost) [reply Pseudocode/Greedy Algorithms/SourceCode.tex]: But what if the input is `[1,2], [1,2]`?

Comment (ghost) [reply Pseudocode/Greedy Algorithms/SourceCode.tex]: The intended solution is 1, (which is obtained by non-decreasing and not monotonic. What do you think?

**Comment by ghost** [comment]
Okay, keep it `monotonic` for now. I'll run some tests locally (and prove the correctness) and update the file later if needed.

File: Pseudocode/Greedy Algorithms/SourceCode.tex (hunk 5 above)
Comment (pjzzz) [reply Pseudocode/Greedy Algorithms/SourceCode.tex]: Don't you think the intended solution will be obtained if sorted monotonically increasing too? I think non-decreasing being a superset will still give the answer but its more accurate to write increasing.

Comment (ghost) [reply Pseudocode/Greedy Algorithms/SourceCode.tex]: But `monotonic` means that if `a` occurs before `b` in sorted order, then `a.finish < b.finish`. However, this isn't the case here. Or am I missing something?

Comment (ghost) [reply Pseudocode/Greedy Algorithms/SourceCode.tex]: Okay, let's keep it `monotonic` for now. I'll try out some examples later and verify them.

**Comment by ghost** [comment]
Please add the 2 blank lines as I mentioned above

File: Pseudocode/Greedy Algorithms/SourceCode.tex (hunk 5 above)
Comment (pjzzz) [reply Pseudocode/Greedy Algorithms/SourceCode.tex]: It's like when we say monotonically increasing if some test case such as yours appears we have to keep one among those with the same finish times. But this raises another question of which among them to select(it should be the one with maximum start time, but the books should clarify)?

Comment (ghost) [reply Pseudocode/Greedy Algorithms/SourceCode.tex]: I think it's better to state that 
> If the `finish` time coincide, then sort by the `start` time

Comment (ghost) [reply Pseudocode/Greedy Algorithms/SourceCode.tex]: And I don't think by monotonically increasing, they mean to remove some entries. (They are only talking about sorting them). But I'm not sure, I'll have to read up the literature.

File: Pseudocode/Greedy Algorithms/Activity Selection/SourceCode.tex [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+\captionsetup[algorithm]{labelformat = algnonumber}
+
+\begin{document}
+
+\begin{algorithm}
+
+  \caption{Maximize the number of activities without overlapping schedules}
+  \begin{algorithmic}[1]
+    \Ensure One Based Indexing for the Array
+    \Statex
```
Comment (pjzzz) [inline Pseudocode/Greedy Algorithms/Activity Selection/SourceCode.tex:27]: Do you mean a blank line before this `\Statex` and after `\Ensure`

File: Pseudocode/Greedy Algorithms/SourceCode.tex (hunk 5 above)
Comment (ghost) [reply Pseudocode/Greedy Algorithms/SourceCode.tex]: If you want, you can add a comment **after** the end of the algorithm(footer) to verify the statement regarding monotonicity

File: Pseudocode/Greedy Algorithms/Activity Selection/SourceCode.tex (hunk 8 above)
Comment (ghost) [reply Pseudocode/Greedy Algorithms/Activity Selection/SourceCode.tex:27]: No, a blank line after the left comment

File: Pseudocode/Greedy Algorithms/Activity Selection/SourceCode.tex (hunk 7 above)
Comment (pjzzz) [reply Pseudocode/Greedy Algorithms/Activity Selection/SourceCode.tex]: Which `\Statex` are we talking about?

File: Pseudocode/Greedy Algorithms/Activity Selection/SourceCode.tex (hunk 8 above)
Comment (ghost) [reply Pseudocode/Greedy Algorithms/Activity Selection/SourceCode.tex:27]: There is already a blank line here

File: Pseudocode/Greedy Algorithms/Activity Selection/SourceCode.tex (hunk 6 above)
Comment (pjzzz) [reply Pseudocode/Greedy Algorithms/Activity Selection/SourceCode.tex]: Ohh so you want me to add a `\Statex` here!! I thought you wanted code indentation.

File: Pseudocode/Greedy Algorithms/Activity Selection/SourceCode.tex (hunk 7 above)
Comment (pjzzz) [reply Pseudocode/Greedy Algorithms/Activity Selection/SourceCode.tex:45]: I have added the `\Statex` right or do I need to add somewhere else?

**Comment by ghost** [comment]
Everything looks correct now. Thank you for your contribution!

**Comment by ghost** [comment]
#13

**Pull request merged**